internal/installer/elasticagent_tar.go

```diff
@@ -0,0 +1,144 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
What is the difference with `tar.go`? I can see both files and at first sight I thought this one would replace the old one, but that's not the case.

Reply:
My intention is to replace the old ones, I wasn't sure if I wanted to do the cleanup in this PR or not, @mdelapenya do you care either way? I can update this PR with the removals if that's ok

Reply:
I'd prefer not having dead code if possible, although I'm not against your proposal ☺️